csv_provider/recorder.py

```diff
@@ -0,0 +1,96 @@
+                        " of the kuksa.val databroker to connect to."
+                        " The default value is 55555", type=int)
+    parser.add_argument("-f", "--file", default="signalsOut.csv", help="This indicates the csv file"
+                        " to write the signals to."
+                        " The default value is signals.csv.")
+    parser.add_argument("-s", "--signals", help="A list of signals to"
+                        " record", nargs='+', required=True)
+    parser.add_argument("-l", "--log", default="INFO", help="This sets the logging level."
+                        " The default value is WARNING.",
+                        choices={"INFO", "DEBUG", "WARNING", "ERROR", "CRITICAL"})
+    return parser
+
+
+async def main():
+    '''entrypoint to the CSV-recorder'''
+    args = init_argparse().parse_args()
+    numeric_value = getattr(logging, args.log.upper(), None)
+    if isinstance(numeric_value, int):
+        logging.basicConfig(encoding='utf-8', level=numeric_value)
+    try:
+        async with VSSClient(args.address, args.port) as client:
+            csvfile = open(args.file, 'w', newline='', encoding="utf-8")
+            signalwriter = csv.DictWriter(csvfile, ['field', 'signal', 'value', 'delay'])
+            signalwriter.writeheader()
+            previous_time = time.time()
+            initial_value = True
+            entries = []
+            for signal in args.signals:
+                entries.append(SubscribeEntry(signal,
+                                              View.FIELDS,
+                                              (Field.VALUE, Field.ACTUATOR_TARGET)))
+            async for updates in client.subscribe(entries=entries):
+                if initial_value:
+                    time_gap = 0.0
+                    initial_value = False
+                else:
+                    current_time = time.time()
+                    time_gap = current_time - previous_time
+                    previous_time = current_time
+                for update in updates:
+                    entry = update.entry
+                    if entry.value is not None:
+                        signalwriter.writerow({'field': 'current',
+                                               'signal': entry.path,
+                                               'value': entry.value.value,
+                                               'delay': time_gap})
+                    if entry.actuator_target is not None:
+                        signalwriter.writerow({'field': 'target',
+                                               'signal': entry.path,
+                                               'value': entry.actuator_target.value,
+                                               'delay': time_gap})
+    except VSSClientError as error:
+        logging.error("There was a problem in the interaction"
+                      " with the Kuksa.val data broker at %s:%s: %s ",
```

Comment:
Replace "data broker" with "Databroker", see https://github.com/eclipse/kuksa.val/wiki/KUKSA.val-Naming-Guidelines

Reply:
I will replace it to align, but personally I prefer 'data broker' since most language checkers ask for this writing and I do not think the term is unique enough to be a proper noun in the context of Eclipse Kuksa.

Reply:
I have no strong opinion on what we use as long as we are consistent. At the moment we mostly use "Databroker", in for example the [README](https://github.com/eclipse/kuksa.val/blob/master/kuksa_databroker/README.md), but if we would agreed that "Data broker" would be better name I have no problem with that, but then we should change it "Data broker" in all free-text-places, and possibly use "data-broker" or "data_broker" as preferred name where blanks cannot be used, like directories and docker containers.